x/alliance/keeper/reward.go

```diff
@@ -131,10 +131,15 @@ func accumulateRewards(latestRewardHistories types.RewardHistories, rewardHistor
 // To calculate the number of rewards claimable, take reward_history * alliance_token_amount * reward_weight
 func (k Keeper) AddAssetsToRewardPool(ctx sdk.Context, from sdk.AccAddress, val types.AllianceValidator, coins sdk.Coins) error {
 	rewardHistories := types.NewRewardHistories(val.GlobalRewardHistory)
+	// We need some delegations before we can split rewards. Else rewards belong to no one and do nothing
+	if len(val.TotalDelegatorShares) == 0 {
+		return nil
+	}
+
 	totalAssetWeight := k.totalAssetWeight(ctx, val)
-	// We need some delegations before we can split rewards. Else rewards belong to no one
 	if totalAssetWeight.IsZero() {
-		return types.ErrZeroDelegations
+		// Do nothing since there are no assets to distribute rewards to
+		return nil
 	}
```

Comment:
I think this case the right behaviour is that we should ignore rewards since there isn't any delegators to distribute the rewards to. Your suggestion could work but right now all uses of this method at the moment need to catch and handle the error. There isn't a place where we don't need to ignore this case. 

I'm thinking this shouldn't be an error that could cause an unexpected panic.